packages/puppeteer-extra/src/index.ts

```diff
@@ -384,6 +406,27 @@ export class PuppeteerExtra implements VanillaPuppeteer {
     }
   }
 
+  /**
+   * Resolves a plugin through Yarn's PnP system.
+   *
+   * Requires Yarn v2.3.0+ due to using the Yarn extension method getAllLocators.
+   *
+   * @private
+   */
+  private resolvePnpPlugin(pluginName: string, requiredBy: string) {
```

Comment:
Wont this be a bit problematic if you have multiple versions of `requiredBy`? It would be safer to have `requiredBy` be a file path

Reply:
Good point, I've changed it to use paths instead. Though I don't really like how I've implemented it, any ideas on how to improve it?

Reply:
Well you can get rid of the PnP specific logic and use
```js
require(require.resolve(pluginName, { paths: [requiredBy] }))
```

Just running a `require(pluginName)` is a problem in `node_modules` as well so if anything this is a bugfix rather than adding PnP support
https://yarnpkg.com/advanced/rulebook#packages-should-only-ever-require-what-they-formally-list-in-their-dependencies

Reply:
Ah, that works much better. Thanks!